File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Mutation.java
package com.example.grts;

import viaduct.engine.api.EngineObjectData;
import viaduct.engine.api.NodeReference;
import viaduct.engine.api.RootFieldReference;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.internal.NodeObjectBase;
import viaduct.java.api.internal.ObjectBase;
import viaduct.java.api.internal.OutputBuilderTypeChecker;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.RootObjectField;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.Arguments;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

@SuppressWarnings("MissingOverride")
public class Mutation extends ObjectBase implements viaduct.java.api.types.Mutation {

    public static final Type<Mutation> Reflection = Type.ofClass(Mutation.class);

    public static final class Fields implements TypeFields<Mutation> {
        private Fields() {}

        public static final Field<Mutation> __typename =
                Field.of("__typename", Reflection);
                public static final CompositeField<Mutation, Order> createOrder =
                                CompositeField.of("createOrder", Reflection, Order.Reflection);

    }

    public Mutation(InternalContext context, EngineObjectData.Sync data) {
        super(context, data);
    }

    @SuppressWarnings("UnusedMethod")
    private Mutation(InternalContext context, Map<String, Object> data) {
        super(context, data, "Mutation");
    }

    private Mutation(InternalContext context, ObjectBase base, Map<String, Object> data) {
        super(context, base, data, "Mutation");
    }

    public Mutation(InternalContext context, RootFieldReference rootFieldReference) {
        super(context, rootFieldReference);
    }
        public Order getCreateOrderOrThrow(String alias) {
            return fetchObject("createOrder", alias, Order.class, Order::new);
        }

        public Order getCreateOrderOrThrow() {
            return fetchObject("createOrder", null, Order.class, Order::new);
        }

        public Order getCreateOrder(String alias) {
            return nullOnDataFailure(() -> fetchObject("createOrder", alias, Order.class, Order::new));
        }

        public Order getCreateOrder() {
            return nullOnDataFailure(() -> fetchObject("createOrder", null, Order.class, Order::new));
        }


    public Builder toBuilder() {
        return new Builder(__context(), toBuilderBase());
    }

    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context), null);
    }

    public static class Builder {
        private final InternalContext __context;
        private final ObjectBase __base;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context, ObjectBase __base) {
            this.__context = __context;
            this.__base = __base;
        }

                public Builder createOrder(Order createOrder) {
                    createOrder = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Mutation",
                            "createOrder",
                            Order.class,
                            createOrder);
                    data.put("createOrder", createOrder);
        return this;
                }


        public Mutation build() {
            return new Mutation(__context, __base, new LinkedHashMap<>(data));
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Mutation.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InternalContext
import viaduct.api.internal.ObjectBase
import viaduct.engine.api.EngineObject
import viaduct.engine.api.EngineObjectData

@OptIn(InternalApi::class)
class Mutation(context: InternalContext, engineObject: EngineObject)
    : ObjectBase(context, engineObject), viaduct.api.types.Object,viaduct.api.types.Mutation
{
     fun getCreateOrderOrThrow(alias: String?): viaduct.api.grts.Order? = TODO()
     fun getCreateOrderOrThrow(): viaduct.api.grts.Order? = TODO()
     fun getCreateOrder(alias: String?): viaduct.api.grts.Order? = TODO()
     fun getCreateOrder(): viaduct.api.grts.Order? = TODO()


    fun toBuilder(): Builder =
        Builder(__context, __engineObject.type, toBuilderEOD())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Mutation =
            Builder(context).apply(block).build()
    }

    class Builder : ObjectBase.Builder<Mutation> {
        constructor(context: ExecutionContext)
            : super(
                context as InternalContext,
                TODO() as graphql.schema.GraphQLObjectType,
                null
            )

        internal constructor(
            context: InternalContext,
            type: graphql.schema.GraphQLObjectType,
            baseEngineObjectData: EngineObjectData.Sync
        ) : super(context, type, baseEngineObjectData)

                  fun createOrder(value: viaduct.api.grts.Order?): Builder = TODO()


        final override fun build(): Mutation = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Mutation> {
        override final val name = "Mutation"
        override final val kcls = viaduct.api.grts.Mutation::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Mutation> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Mutation> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Mutation.Reflection)

            final val createOrder: viaduct.api.reflect.CompositeField<viaduct.api.grts.Mutation, viaduct.api.grts.Order> =
                viaduct.api.internal.CompositeFieldImpl("createOrder", viaduct.api.grts.Mutation.Reflection, viaduct.api.grts.Order.Reflection)

    }

}